### PR TITLE
Simplify ensemble retriever

### PR DESCRIPTION
**Pull request by ahmed-moubtahij**
- **Description:** code simplification to improve readability and remove unnecessary memory allocations.
  - **Tag maintainer**: @baskaryan, @eyurtsev, @hwchase17.

### CODE REVIEW

File: libs/core/langchain_core/utils/utils.py [hunk 1]
```diff
@@ -38,6 +38,14 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
 
     return decorator
 
+T = TypeVar('T')
+H = TypeVar('H', bound=Hashable)
+def unique_by_key(iterable: Iterable[T], key: Callable[[T], H]) -> Iterator[T]:
```
Comment (hwchase17) [inline libs/core/langchain_core/utils/utils.py:43]: maybe just worth being local to ensemble retriever until we use in multiple places?

Comment (Ayenem) [reply libs/core/langchain_core/utils/utils.py]: Done!

Comment (Ayenem) [reply libs/core/langchain_core/utils/utils.py]: @hwchase17 Any more changes you'd like to see?

Comment (Ayenem) [reply libs/core/langchain_core/utils/utils.py]: @hwchase17 I apparently needed to run `make lint` at the langchain_core level as well (root didn't detect the lint fails on CI), it should be ok now.

**Comment by Ayenem** [comment]
@hwchase17 @baskaryan Looks good to go?

**Comment by Ayenem** [comment]
make format and make lint pass locally but fail on CI (`make lint #3.8`):

> Error: langchain/retrievers/ensemble.py:5:1: I001 Import block is un-sorted or un-formatted
> Error: langchain/retrievers/ensemble.py:9:89: E501 Line too long (89 > 88)
> make: *** [Makefile:62: lint_package] Error 1
> Error: Process completed with exit code 2.

I don't know how to fix this.

**Comment by cold-eye** [comment]
> make format and make lint pass locally but fail on CI (`make lint #3.8`):
> 
> > Error: langchain/retrievers/ensemble.py:5:1: I001 Import block is un-sorted or un-formatted
> > Error: langchain/retrievers/ensemble.py:9:89: E501 Line too long (89 > 88)
> > make: *** [Makefile:62: lint_package] Error 1
> > Error: Process completed with exit code 2.
> 
> I don't know how to fix this.

how did you fix? I am meeting the same errors

**Pull request merged**